### ISSUE TITLE
Inconsistent performance under PyTorch=1.12.1 Functorch=0.2.1 and PyTorch=1.13.1 Functorch=0.13.1

### ISSUE THREAD

**Issue by Jinxi-Li** [martenlienen/torchode#8]
Hi! I don't know where the problem is and only found this behavior by chance. 

1. I first run the model on one machine with PyTorch=1.12.1, cuda=11.6 and Functorch=0.2.1, the performance is normal (blue loss in the below example). 
2. Then I installed the torchode on another pc, (which pre-installed pytorch=1.12.1 but was reinstalled to pytorch=1.13.1 by functorch when installing torchode), and ran the same code again, the performance changed a lot. It becomes hard to converge (orange loss in the example below). 
3. After I checked the environment, and reinstalled the pytorch and functorch back to 1.12.1 and 0.2.1 manually, the performance get normal again. (yellow loss in the example below).

I don't know what the error is, and it's hard for me to find a toy example about this. Therefore I recommend to change the dependency of the packages in `pip` directly to functorch 0.2.1, in order not to install functorch 0.13.1 which forces the pytorch to be 1.13.1, in case someone meets similar problem as mine. 
<img width="289" alt="loss_for_different_version" src="https://user-images.githubusercontent.com/58667274/212848471-8920a7e0-9ac0-4136-a7aa-03180a4fab6f.png">

**Comment by martenlienen** [comment]
Hi! That is really weird and an issue that I have not observed myself yet. If you manage to distill the issue into some example at some point, I would be grateful and will look into it! Off the top of my head, I cannot think of any difference between pytorch 1.12 and 1.13 that could explain this difference. Is it at all possible that the difference between the versions comes from the model that you put into torchode, that you use some function that changed its behavior slightly between 1.12 and 13?

Also, I am surprised that installing torchode would upgrade your pytorch. I thought that pip would install a version of functorch that is compatible with any already installed version of pytorch.

**Comment by Jinxi-Li** [comment]
> Hi! That is really weird and an issue that I have not observed myself yet. If you manage to distill the issue into some example at some point, I would be grateful and will look into it! Off the top of my head, I cannot think of any difference between pytorch 1.12 and 1.13 that could explain this difference. Is it at all possible that the difference between the versions comes from the model that you put into torchode, that you use some function that changed its behavior slightly between 1.12 and 13?
> 
> Also, I am surprised that installing torchode would upgrade your pytorch. I thought that pip would install a version of functorch that is compatible with any already installed version of pytorch.

I was also surprised that my pytorch was upgraded automatically, maybe I didn't install the functorch before. So at first I thought it was my model's problem and put a lot of time dealing with it. When I turned off the ode module, the performance becomes consistent cross different pytorch versions. So I guess the problem may be due to different functorch version. But the version of functorch is tied to the version of the pytorch. 

For your information, my network is mainly using mlp and `F.grid_sample`, I don't if this can help.

**Comment by martenlienen** [comment]
What do you mean with "turned off the ode module"? Sounds like your model does not use any esoteric functions, so it is probably not the issue. Are you using `BacksolveAdjoint`? Because that is the only thing that actually uses functorch. If you do, please try `AutoDiffAdjoint` that just backprops via pytorch's basic AD functionality.

**Comment by Jinxi-Li** [comment]
By saying "turned off the ode module" I mean I don't optimize the ode model, only training the decoder. And I'm using `AutoDiffAjoint`. 

My ode codes is like:
```
term = to.ODETerm(model, with_args=False)
step_method = to.Dopri5(term=term)
step_size_controller = to.IntegralController(atol=atol, rtol=rtol, term=term)
solver = to.AutoDiffAdjoint(step_method, step_size_controller)
```

**Comment by martenlienen** [comment]
Hm, that looks fine to me. If you ever get some evidence as to what the cause might be, I would be very interested.

**Comment by Jinxi-Li** [comment]
Let me close this issue for now since I can't find another case has the same problem. Once I can locate the problem more accurately, I'll reopen it.
